client/src/pages/trainer/courses/create/index.tsx

```diff
@@ -58,15 +60,22 @@ const Create: FC = () => {
         }
         return await trigger(['name', 'category']);
       case 2:
-        /*
-            Please collect the form inputs in here, since user is at the preview section, crouseSlice should have all the necessary info we need to save in db
-
-            Please reset the courseSlice and stepperSlice to initial values once the from values are saved in db
-        */
-        return await push('/trainer/courses');
+        try {
+          const data = {
+            ...values,
+            category: values.category.map(({ id }) => id),
+          };
+          await API.post('course/', data);
+          void push('/trainer/courses');
+          dispatch(reset());
+        } catch (error) {
+          console.error('Error saving course data:', error);
+        }
```

Comment:
```suggestion
        try {
          const data = {
            ...values,
            category: values.category.map(({ id }) => id),
          };
          await API.post('course/', data);
          await push('/trainer/courses');
          dispatch(reset());
        } catch (error) {
        }
```

1. Please utilize RTK query for post requests as well.
2. We need toast message for success and failure so please remove the comment for error and instead implement toast message
3. Please use `await` for `push('/trainer/courses');`